x-pack/legacy/plugins/actions/server/lib/task_runner_factory.ts

```diff
@@ -93,6 +99,17 @@ export class TaskRunnerFactory {
             executorResult.data,
             executorResult.retry == null ? false : executorResult.retry
           );
+        } else if (executorResult.status === 'ok') {
```

Comment:
If the action error'ed, but indicated a retry, we don't retry forever, do we?  Assuming we have some limit on retries, seems like we'd need same logic where we figure out they're past the retry limit.  Fine handling that as a new issue, obviously we will catch most of the cleanup cases here.

Reply:
> If the action error'ed, but indicated a retry, we don't retry forever, do we?

This depends on the action type. By default I think we only attempt once but you can customize the retry logic per action type.

> Assuming we have some limit on retries, seems like we'd need same logic where we figure out they're past the retry limit. Fine handling that as a new issue, obviously we will catch most of the cleanup cases here.

A new issue for this is good, I'll go ahead and create one. It falls under the same issue as failed tasks in task manager, they remain there forever at this time.

Reply:
Created #55340.